core/bin/snapshots_creator/src/main.rs

```diff
@@ -0,0 +1,306 @@
+mod chunking;
+
+use anyhow::Context as _;
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use prometheus_exporter::PrometheusExporterConfig;
+use std::cmp::max;
+use std::future::Future;
+use std::pin::Pin;
+use std::time::Duration;
+use tokio::sync::watch;
+use tokio::sync::watch::Receiver;
+use vise::Unit;
+use vise::{Buckets, Gauge, Histogram, Metrics};
+use zksync_config::configs::PrometheusConfig;
+use zksync_config::{PostgresConfig, SnapshotsCreatorConfig};
+
+use crate::chunking::get_chunk_hashed_keys_range;
+use zksync_dal::ConnectionPool;
+use zksync_env_config::object_store::SnapshotsObjectStoreConfig;
+use zksync_env_config::FromEnv;
```

Comment:
Bikeshedding: I understand that we don't enforce or even have a consistent import style, but IMO we should *at least* ensure at most one import entry for each crate. Not blocking obviously, so feel free to ignore.